lib/Sema/TypeCheckType.cpp

```diff
@@ -1325,6 +1260,26 @@ resolveTopLevelIdentTypeComponent(TypeResolution resolution,
     if (options.contains(TypeResolutionFlags::SilenceErrors))
       return ErrorType::get(ctx);
 
+    if (comp->getIdentifier() == ctx.Id_Self &&
+        !isa<GenericIdentTypeRepr>(comp)) {
+      DeclContext *nominalDC = nullptr;
+      NominalTypeDecl *nominal = nullptr;
+      auto dc = resolution.getDeclContext();
+      if ((nominalDC = dc->getInnermostTypeContext()) &&
+          (nominal = nominalDC->getSelfNominalTypeDecl())) {
+        // Attempt to refer to 'Self' within a non-protocol nominal
+        // type. Fix this by replacing 'Self' with the nominal type name.
```

Comment:
"Fix" suggests the code is invalid and we're about to offer a fixit. But in fact it is valid

Reply:
removed obsolete comment